lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1598 @@
+      - SUSE Linux (Amazon VPC)
+      - Windows
+    description:
+      - (String) Operation system type._
+    required: true
+
+  rancher:
+    description:
+      - (Object) The Rancher integration configuration.;
+        Expects the following keys -
+        access_key (String),
+        secret_key (String),
+        master_host (String)
+    required: false
+
+  right_scale:
+    description:
+      - (Object) The Rightscale integration configuration.;
+        Expects the following keys -
+        account_id (String),
+        refresh_token (String)
+    required: false
+
+  risk:
+    description:
+      - (Integer) required if on demand is not set. The percentage of Spot instances to launch (0 - 100).
+    required: false
+
+  roll_config:
+    description:
+      - (Object) Roll configuration.;
+        If you would like the group to roll after updating, please use this feature.
+        Accepts the following keys -
+        batch_size_percentage(Integer, Required),
+        grace_period - (Integer, Required),
+        health_check_type(String, Optional),
+    required: false
+
+  scheduled_tasks:
+    description:
+      - (List of Objects) a list of hash/dictionaries of scheduled tasks to configure in the elastigroup;
+        '[{"key":"value", "key":"value"}]';
+        keys allowed are -
+        adjustment (Integer),
+        scale_target_capacity (Integer),
+        scale_min_capacity (Integer),
+        scale_max_capacity (Integer),
+        adjustment_percentage (Integer),
+        batch_size_percentage (Integer),
+        cron_expression (String),
+        frequency (String),
+        grace_period (Integer),
+        task_type (String, required),
+        is_enabled (Boolean)
+    required: false
+
+  security_group_ids:
+    description:
+      - (List of Strings) One or more security group IDs. ;
+        In case of update it will override the existing Security Group with the new given array
+    required: true
+
+  shut_down_script:
+    description:
+      - (String) The Base64-encoded shutdown script that executes prior to instance termination.
+        Encode before setting.
+    required: false
+
+  signals:
+    description:
+      - (List of Objects) a list of hash/dictionaries of signals to configure in the elastigroup;
+        keys allowed are -
+        name (String, required),
+        timeout (Integer)
+    required: false
+
+  spin_up_time:
+    description:
+      - (Integer) spin up time, in seconds, for the instance
+    required: false
+
+  spot_instance_types:
+    description:
+      - (List of Strings) Spot instance type that will be provisioned.
+    required: true
+
+  state:
+    choices:
+      - present
+      - absent
+    description:
+      - (String) create or delete the elastigroup
+    required: false
+
+  tags:
+    description:
+      - (Dictionary) a dictionary of tags to configure in the elastigroup. Please specify list of keys and values (key colon value);
```

Comment:
Tags is compatible with ec2_asg module and now takes a **list**, so this should be updated.